dask/dataframe/io/parquet/arrow.py

```diff
@@ -1517,7 +1520,49 @@ def _read_table(
 
         This method is overridden in `ArrowLegacyEngine`.
         """
-        if isinstance(path_or_frag, pa_ds.ParquetFileFragment):
+
+        # Check if we have partitioning information.
+        # Will only have this if the engine="pyarrow-dataset"
+        partitioning = kwargs.pop("partitioning", None)
+
+        # Check if we need to generate a fragment for filtering.
+        # We only need to do this if we are applying filters to
+        # columns that were not already filtered by "partition".
+        if partitioning and filters:
+            partition_cols = (
+                set([v[0] for v in flatten(partition_keys, container=list) if len(v)])
+                if partition_keys
+                else set()
+            )
+            filtered_cols = (
+                set([v[0] for v in flatten(filters, container=list) if len(v)])
+                if filters
+                else set()
+            )
+            need_fragment = bool(filtered_cols - partition_cols)
+        else:
+            need_fragment = False
```

Comment:
Only want to generate a fragment object if we need to apply a row-wise filter.  We do **not** need to regenerate the fragment if we are only filtering by "partitioned" columns (since they were already fltered).